src/hpack/table.rs

```diff
@@ -125,7 +125,7 @@ impl Table {
             Indexed(idx, ..) => idx,
             Name(idx, ..) => idx,
             Inserted(idx) => idx + DYN_OFFSET,
-            InsertedValue(idx, _) => idx,
+            InsertedValue(_name_idx, slot_idx) => slot_idx + DYN_OFFSET,
```

Comment:
This makes the next header (that has no name, so using this index) point at the new index, not at the old name index, which may have been evicted.